src/customio.cpp

```diff
@@ -0,0 +1,242 @@
+    pInstance->m_iMaxHealth(clamp(Q_atoi(vecArgs[1].c_str()), 1, 99999999));
+    const int m_iMaxHealth = pInstance->m_iMaxHealth;
+    Message("SetMaxHealth %d for %s", m_iMaxHealth, pInstance->GetName());
+}
+
+static void AddOutputCustom_Health(Z_CBaseEntity*                  pInstance,
+                                   CEntityInstance*                pActivator,
+                                   CEntityInstance*                pCaller,
+                                   const std::vector<std::string>& vecArgs)
+{
+    const auto value = clamp(Q_atoi(vecArgs[1].c_str()), 1, 99999999);
+    pInstance->m_iHealth(value);
+    const int m_iHealth = pInstance->m_iHealth;
+    Message("SetHealth %d for %s", m_iHealth, pInstance->GetName());
+}
+
+static void AddOutputCustom_MoveType(Z_CBaseEntity*                  pInstance,
+                                     CEntityInstance*                pActivator,
+                                     CEntityInstance*                pCaller,
+                                     const std::vector<std::string>& vecArgs)
+{
+    static Vector stopVelocity(0, 0, 0);
+    const auto    value = clamp(Q_atoi(vecArgs[1].c_str()), MOVETYPE_NONE, MOVETYPE_LAST);
+    if (const auto type = static_cast<MoveType_t>(clamp(value, MOVETYPE_NONE, MOVETYPE_LAST));
+        type == MOVETYPE_NONE || type == MOVETYPE_WALK)
+    {
+        pInstance->SetMoveType(type);
+        if (type == MOVETYPE_NONE) // stop manually!
+            pInstance->Teleport(nullptr, nullptr, &stopVelocity);
+        Message("SetMoveType %d for %s", type, pInstance->GetName());
+    }
+}
+
+static void AddOutputCustom_EntityTemplate(Z_CBaseEntity*                  pInstance,
+                                           CEntityInstance*                pActivator,
+                                           CEntityInstance*                pCaller,
+                                           const std::vector<std::string>& vecArgs)
+{
+    if (strcmp(pInstance->GetClassname(), "env_entity_maker") == 0)
+    {
+        const auto pEntity = reinterpret_cast<CEnvEntityMaker*>(pInstance);
+        const auto pValue  = g_pEntitySystem->AllocPooledString(vecArgs[1].c_str());
+        pEntity->m_iszTemplate(pValue);
+        Message("Set EntityTemplate to %s for %s\n", pValue.String(), pInstance->GetName());
+    }
+    else
+        Message("Only env_entity_maker is supported\n");
+}
+
+static void AddOutputCustom_BaseVelocity(Z_CBaseEntity*                  pInstance,
+                                         CEntityInstance*                pActivator,
+                                         CEntityInstance*                pCaller,
+                                         const std::vector<std::string>& vecArgs)
+{
+    const Vector velocity(clamp(Q_atof(vecArgs[1].c_str()), -3500.f, 3500.f),
+                          clamp(Q_atof(vecArgs[2].c_str()), -3500.f, 3500.f),
+                          clamp(Q_atof(vecArgs[3].c_str()), -3500.f, 3500.f));
```

Comment:
This should be clamped to 4096 since that's the actual limit of velocity vectors in CS2.

Reply:
that old values are came from cs:go.